### DISCUSSION TITLE
How to pass image masks to 'ns-process-data images'?

### DISCUSSION THREAD

**Discussion by andynewman10** [nerfstudio-project/nerfstudio#3097] [General]
I have nerfstudio running and it is working fine. I am using the following commands:

```
ns-process-data images --data mydataset --output-dir mydatasetprocessed
cd mydatasetprocessed
ns-train
```

All is running well however I cannot figure out how to pass per-image masks, just like the way I am doing it manually using the COLMAP GUI. This is a blocking issue for me.

When using the COLMAP GUI, I can specify the folder for images (`images`) and another folder for corresponding masks (`masks`). It works fine.

Where is `ns-process-data images` expected to find image masks? I read the documentation and it seems to be supported, but I am not sure if it really is (EDIT: the following issue talks about a Mustard dataset, with masks, but unfortunately the link to the dataset is now dead: https://github.com/nerfstudio-project/nerfstudio/issues/1498#issue-1599115766)

Considering `ns-process-data images` may not be supporting masks at all, I tried to

- manually run COLMAP. This generates the `sparse/0` folder with the bin files, as well as the database.db and project.ini files.
- instruct `ns-process-data images` to use existing COLMAP output as-is, and just generate the `transforms.json` file, without which training cannot take place. I saw there is a `--skip-colmap` option for this purpose.

Unfortunately, `ns-process-data images --skip-colmap` does not want to proceed because it does not find the images_2, images_4 and images_8 subfolders, that are automatically generated under normal circumstances (when --skip-colmap is not used).

**Comment by Marlazzar** [comment]
i have the same problem
